### ISSUE TITLE
update contribution page

### ISSUE THREAD

**Issue by lemming622** [iglance/iGlance#190]
I am having problems starting to clone the iGlance repository.  Would someone be able to update the contribution steps to include the git clone command:<br>`git clone --recurse-submodules <your-git-repo-here>`<br>?
It took me a moment to find out why the pod command wasn't working and it was due to the submodules not being cloned as well.
After solving that issue I am now running into an issue with not being able to clone the submodules since I don't have the correct access rights.
```
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

Would a more detailed (step-by-step) set of instructions be created?

**Comment by D0miH** [comment]
Hey @lemming622,
thanks for reaching out. What did you use to clone the main iGlance repository? Did you clone the repository through SSH or through HTTPS?

**Comment by Moneypulation** [comment]
Can confirm the recursive submodule problem. I ran into it as well when I initially cloned iGlance without the --recursive flag. Therefore, we should mention to use it in to guideline imo.

I assume the access right problem is due to the fact that your ssh key isn't configured correctly. But you can also clone the repo through https

**Comment by D0miH** [comment]
The problem here is that the submodules urls in the main repo are configured to be ssh. Therefore, if you clone the main repo using HTTPS because you don't want to authenticate to GitHub you will get problems when trying to clone the submodules.

Therefore, best option to circumvent all those problems is probably to just upload a ssh key to GitHub  and clone the repository using
```bash
git clone --recurs-submodules git@github.com:iglance/iGlance.git
```
This should clone the main repo as well as all submodules correctly.

But I totally agree that we should add a step-by-step guide to make it easier for newcomers to also participate.

**Comment by D0miH** [comment]
@lemming622 One additional step you have to take is that you have to install [Carthage](https://github.com/Carthage/Carthage), switch into the `<repo-root>/iGlance/iGlance` directory and execute the command `carthage update`. This is because I added a library that was only available through carthage.

If you have any additional difficulties during the setup feel free to ask here. I will update a step-by-step guide on how to setup the project locally asap.

**Comment by lemming622** [comment]
@D0miH, I was using the HTTPS cloning.  I have uploaded an ssh key so I could do a clone using ssh.  Which did work.  Thank you for helping debug that.

Now the next spot that I'm having issues with is the Carthage update.  Here are my commands and outputs.  I'm not familiar with Carthage so I don't know how to fix this one.<br>
```
(base) me@My-MacBook-Pro iGlance % pwd            
/Users/me/Downloads/git/iGlance/iGlance
(base) me@My-MacBook-Pro iGlance % carthage update
Failed to read file or folder at /Users/me/Downloads/git/iGlance/iGlance/Cartfile: Error Domain=NSCocoaErrorDomain Code=260 "The file “Cartfile” couldn’t be opened because there is no such file." UserInfo={NSFilePath=/Users/me/Downloads/git/iGlance/iGlance/Cartfile, NSUnderlyingError=0x7f99e5dac610 {Error Domain=NSPOSIXErrorDomain Code=2 "No such file or directory"}}
```

**Comment by D0miH** [comment]
Iirc there is no `Cartfile` in the directory where you run the `carthage update` command.
This can have two reasons:
1. Either you are on the master branch which is behind the development branch and therefore has no Cartfile (yet).
2. Or you are in the wrong directory and have to switch into the correct one. The `Cartfile` is in the same directory as the `iGlance.xcodeproj` file (note that it is the iGlance<b>.xcodeproj</b> not the iGlance<b>.xcworkspace</b> file).

Sorry for the inconvenience during the setup. Quite a lot has changed since I wrote that contribution guide and as you probably know it is like the saying "documentation is the last thing you write and the first thing that gets outdated" 😉
However, I will update the contribution guide in the next few days.

**Comment by lemming622** [comment]
@D0miH, thank you again for the debugging help on this.  Yes I had checked out the master vs the development branch.  I have since changed branches.  The next issue I had was with Carthage running `carthage update`.  I found the solution to it.  Turns out there is an issue with XCode if you have not enabled command line processing.  <br>Here is a link describing [the issue and solution](https://stackoverflow.com/questions/40743713/command-line-tool-error-xcrun-error-unable-to-find-utility-xcodebuild-n).  I'm including that for part of the rewrite.
